src/packages/record-screen/index.ts

```diff
@@ -0,0 +1,124 @@
+import writeToStore from '@/packages/write-to-store';
+import Features from '@/packages/features';
+import FeatureToggleStore from '@/packages/feature-toggle-store';
+
+export interface ScreenRecorderOptions {
+  videoName?: string;
+  encodingType?: string,
+}
+
+export class ScreenRecorder<T extends ScreenRecorderOptions> {
+  private recorder;
+  private data;
+  private options: T;
+
+  constructor(options: T) {
+    this.options = options;
+  }
+
+  public async start(): Promise<void> {
+    if (!this.isEnabled() || !this.isBrowserCompatible()) {
+      return;
+    }
+
+    const constraints = { video: { mediaSource: "screen" } };
+
+    const stream = await this.getDisplayMedia(constraints);
+
+    this.recorder = new window['MediaRecorder'](stream);
```

Comment:
maybe add a `// @ts-ignore` before it? so at least the intent of this can be a bit more explicit :P